app/services/search_service.rb

```diff
@@ -319,4 +319,24 @@ def one_of_regex(values)
   def letter?(char)
     char =~ /[[:alpha:]]/
   end
+
+  def cast(val)
+    return val unless val.is_a?(String)
+
+    begin
+      return Integer(val)
```

Comment:
Does elasticsearch actually care about the difference between an int vs float if it's going across as json either way?

Reply:
Good question. I find that this approach mostly just guarantees that we don't lose precision anywhere down the line. I expect the downstream JSON encoding behavior via Searchkick to be smart enough to handle the parsed types better than as simple strings.